plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnector.java

```diff
@@ -970,12 +975,23 @@ protected void doPush(final String repository,
                     @Override
                     public void run() {
                         try {
+                            // TODO add tag into pattern (now it is always null)
+                            //String pattern = tag + ": digest: (?<" + DIGEST_GROUP + ">sha256:[a-z0-9]{64}) size: \\d+";
+                            Pattern digestPattern = Pattern.compile("digest: (?<" + DIGEST_GROUP + ">sha256:[a-z0-9]{64}) size: \\d+");
+                            Matcher digestMatcher;
                             ProgressStatus progressStatus;
                             while ((progressStatus = progressReader.next()) != null && exceptionHolder.get() == null) {
                                 progressMonitor.updateProgress(progressStatus);
                                 if (progressStatus.getError() != null) {
                                     exceptionHolder.set(progressStatus.getError());
                                 }
+                                String status = progressStatus.getStatus();
+                                if (status != null) {
+                                    digestMatcher = digestPattern.matcher(status);
```

Comment:
It will be executed hundreds of times or even more. Can we avoid usage of pattern here? It is quite expensive.